thorlcr/thorutil/thmem.cpp

```diff
@@ -588,9 +586,22 @@ inline bool CThorExpandingRowArray::_ensure(rowidx_t requiredRows, unsigned maxS
 CThorExpandingRowArray::CThorExpandingRowArray(CActivityBase &_activity, IRowInterfaces *_rowIf, bool _allowNulls, StableSortFlag _stableSort, bool _throwOnOom, rowidx_t initialSize)
     : activity(_activity)
 {
+    stableTable = NULL;
+    rows = NULL;
+    maxRows = 0;
+    numRows = 0;
+    rowManager = activity.queryJob().queryRowManager();
+    throwOnOom = false;
     setup(_rowIf, _allowNulls, _stableSort, _throwOnOom);
     setDefaultMaxSpillCost(roxiemem::SpillAllCost);
-    init(initialSize);
+    if (initialSize)
```

Comment:
I don't think so.. it _could_ for common code reasons, but more efficient/simpler code to avoid checks/callbacks here in constructor where there is no contention/locking required.